client/blocks/site/index.jsx

```diff
@@ -194,7 +196,16 @@ class Site extends Component {
 						size={ this.props.compact ? 24 : this.props.isReskinned ? 50 : 32 }
 					/>
 					<div className="site__info">
-						<div className="site__title">{ site.title }</div>
+						{ ! this.props.showChevronIcon ? (
+							<div className="site__title">{ site.title }</div>
+						) : (
+							<div className="site__title-with-chevron-icon">
+								<span className="site__title">{ site.title }</span>
+								<span className="site__title-chevron-icon">
+									<Gridicon icon="chevron-right" size={ 18 } />
+								</span>
+							</div>
```

Comment:
The change in the elements was needed to ensure that the fade-out effect would continue working as expected for long site names.